db/model/Post.ts

```diff
@@ -72,3 +99,191 @@ export const getPostEnrichedBySlug = async (
     if (!post) return undefined
     return parsePostRow(post)
 }
+
+export const getPostEnrichedById = async (
+    id: number
+): Promise<DbEnrichedPost | undefined> => {
+    const post = await getPostRawById(id)
+    if (!post) return undefined
+    return parsePostRow(post)
+}
+
+export const getFullPostBySlugFromSnapshot = async (
+    slug: string
+): Promise<FullPost> => {
+    const postEnriched = await getPostEnrichedBySlug(slug)
+    if (
+        !postEnriched?.wpApiSnapshot ||
+        !snapshotIsPostRestApi(postEnriched.wpApiSnapshot)
+    )
+        throw new JsonError(`No page snapshot found by slug ${slug}`, 404)
+
+    return getFullPost(postEnriched.wpApiSnapshot)
+}
+
+export const getFullPostByIdFromSnapshot = async (
+    id: number
+): Promise<FullPost> => {
+    const postEnriched = await getPostEnrichedById(id)
+    if (
+        !postEnriched?.wpApiSnapshot ||
+        !snapshotIsPostRestApi(postEnriched.wpApiSnapshot)
+    )
+        throw new JsonError(`No page snapshot found by id ${id}`, 404)
+
+    return getFullPost(postEnriched.wpApiSnapshot)
+}
+
+export const isPostSlugCitable = async (slug: string): Promise<boolean> => {
+    const entries = SiteNavigationStatic.categories
+    return entries.some((category) => {
+        return (
+            category.entries.some((entry) => entry.slug === slug) ||
+            (category.subcategories ?? []).some(
+                (subcategory: CategoryWithEntries) => {
+                    return subcategory.entries.some(
+                        (subCategoryEntry) => subCategoryEntry.slug === slug
+                    )
+                }
+            )
+        )
+    })
+}
+
+export const getPostsFromSnapshots = async (
+    postTypes: string[] = [WP_PostType.Post, WP_PostType.Page],
+    filterFunc?: FilterFnPostRestApi
+): Promise<PostRestApi[]> => {
+    const rawPosts: DbRawPost[] = (
+        await db.knexInstance().raw(
+            `
+                SELECT * FROM ${postsTable}
+                WHERE status = "publish"
+                AND type IN (?)
+                ORDER BY JSON_UNQUOTE(JSON_EXTRACT(wpApiSnapshot, '$.date')) DESC;
```

Comment:
why not `wpApiSnapshot ->> '$.date'` here?

Reply:
who knows? 😅 Thanks!